apps/nextjs/README.md

```diff
@@ -0,0 +1,111 @@
+# Plone on NextJS
+
+This is a PoC of a [NextJS](https://nextjs.org) app using the app router and the `@plone/client` and `@plone/components` library. This is intended to serve as playground for the development of both packages and as demo of Plone using NextJS.
+
+## Development
+
+To start, from the root of the monorepo:
+
+```shell
+pnpm install
+pnpm --filter plone-nextjs run dev
+```
+
+A running Plone backend instance should be running:
+
+```shell
+make start-backend-docker
+```
+
+## Deployment at Vercel
+
+### Vercel
+
+We are introducing an environment variable `API_SERVER_URL`.
+We have to create this environment variable in the Vercel deployment's control panel, specifying the URL where your backend API server is deployed and the route where the api is located (see next section). (eg. API_SERVER_URL=https://my_server_DNS_name/api).
+
+### Application rewrite config
+
+In order to avoid CORS and maintain the server counterpart private, our NextJS app should have a rewrite and should be configured as follows:
+
+```
+const nextConfig = {
+  // Rewrite to the backend to avoid CORS
+  async rewrites() {
+    const apiServerURL =
+      process.env.API_SERVER_URL ||
+      'http://localhost:8080/Plone/%2B%2Bapi%2B%2B';
+
+    return [
+      {
+        source: '/\\+\\+api\\+\\+/:slug*',
+        destination:
+          `${apiServerURL}/VirtualHostBase/https/${process.env.NEXT_PUBLIC_VERCEL_URL}%3A443/Plone/%2B%2Bapi%2B%2B/VirtualHostRoot/:slug*`,
+      },
+    ];
+  },
+};
+```
+
+Plone Client is using the `++api++` prefix as default, so we should create a redirect in our app pointing to the api server, but using the traditional Plone's VHM config.
+
+NextJS rewrites are picky on the `destination` field because the library that is used for it does not support URLS with regex operators.
+Therefore, we can't use the usual `++api++` route for the rewrite.
+This will allow us to infer the current server URL (even in deployed branches/PRs) without touching the rewrite rules.
+We will fallback to configure a `api` route in our reverse proxy of choice.
+
+### Plone backend
+
+You have to deploy the Plone backend elsewhere, since Vercel is serverless oriented.
+We need to setup the rewrite rule in NextJS's `rewrite` feature as shown in the previous section.
+
+We will fallback to configure a `api` route in our reverse proxy of choice.
+
+If we are using `traefik`:
+
+```
```

Comment:
This needs a syntax language. Not sure what it is because I don't see a file extension to make a guess.

Reply:
Done